opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/aggregation.py

```diff
@@ -80,14 +85,26 @@ class AggregationTemporality(IntEnum):
 
 
 class _Aggregation(ABC, Generic[_DataPointVarT]):
-    def __init__(self, attributes: Attributes):
+    def __init__(
+        self,
+        attributes: Attributes,
+        reservoir_factory: ExemplarReservoirFactory,
+    ):
         self._lock = Lock()
         self._attributes = attributes
+        self._reservoir = reservoir_factory()
         self._previous_point = None
 
-    @abstractmethod
```

Comment:
I think we still want to enforce child classes to need to have to implement `aggregate`. You could probably refactor out the reservoir logic into a helper method.